source/includes/connection-snippets/x509-snippets/c-connection.c

```diff
@@ -0,0 +1,30 @@
+// begin x509 connection
+#include <mongoc/mongoc.h>
+
+int
+main (int argc, char *argv[])
+{
+   mongoc_database_t *database;
+   mongoc_client_t *client;
+
+   mongoc_init ();
+
+   const mongoc_ssl_opt_t *ssl_default = mongoc_ssl_opt_get_default ();
```

Comment:
I assume this code was inspired by the [Connecting to a server over SSL](http://mongoc.org/libmongoc/current/advanced-connections.html#connecting-to-a-server-over-ssl) docs, as the [X509 examples in the libmongoc docs](http://mongoc.org/libmongoc/current/authentication.html#x-509-certificate-authentication) don't use `mongoc_ssl_opt_get_default` at all.

AFAIK, the only purpose of copying in `mongoc_ssl_opt_get_default ()` instead of starting with totally empty SSL opts is that we make it possible to pick up a custom CA file/directory that might have been configured when libmongoc was originally compiled. The X509 example overrides the CA _file_, but I suppose there may still be a use case for inheriting some default CA _directory_. No harm in keeping this as-is, I just wanted to example what's going on.

cc: @bazile-clyde @kevinAlbs

Reply:
Currently I think `const mongoc_ssl_opt_t *ssl_default = mongoc_ssl_opt_get_default()` is equivalent to `mongoc_ssl_opt_t ssl_default = {0}`.

But there's an old TODO here: https://github.com/mongodb/mongo-c-driver/blob/1.15.0/src/libmongoc/src/mongoc/mongoc-ssl.c/#L37-L44 indicating we could change the behavior of `mongoc_ssl_opt_get_default`. I'm not sure how useful that would be, or if that's something we really plan to do though.

Reply:
Ah, I see that `MONGOC_SSL_DEFAULT_TRUST_FILE` and `MONGOC_SSL_DEFAULT_TRUST_DIR` are always defined as `NULL` if undefined at compile time. And I found no references to them in CMake so I assume they are simply unused and reserved for future use.

@kevinAlbs: In that case, do you suppose it makes sense to omit `mongoc_ssl_opt_get_default ()` from this example entirely?

Reply:
Per this thread  https://github.com/mongodb/docs-ecosystem/pull/573/files/c5875fa5382cf3e2ccbf27cc2d8843d3359031e1#r348478333
I'd prefer removing the reference to mongoc_ssl_opt_t entirely and use the URI opts instead.

Reply:
@kevinAlbs To speed this along, would you mind modifying this file to the preferred form and sending that over to @mongomoe so he can copy it into this PR?